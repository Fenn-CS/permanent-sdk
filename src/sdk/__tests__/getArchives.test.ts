import nock from 'nock';
import { getArchives } from '..';

describe('getArchives', () => {
  it('should return an array of archive VOs', async () => {
    nock('https://permanent.local')
      .get('/api/archive/getAllArchives')
      .replyWithFile(
        200,
        `${__dirname}/replies/getArchives/multipleArchives.json`,
        {
          'Content-Type': 'application/json',
        },
      );

    const archives = await getArchives({
      bearerToken: '12345',
      baseUrl: 'https://permanent.local/api',
    });

    expect(archives).toMatchSnapshot();
  });
});
