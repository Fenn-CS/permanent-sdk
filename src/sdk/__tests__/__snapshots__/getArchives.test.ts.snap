// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getArchives should return an array of archives 1`] = `
Array [
  Object {
    "createdAt": 2022-06-23T18:33:43.000Z,
    "id": 1,
    "name": "dan",
    "updatedAt": 2022-06-23T18:33:57.000Z,
  },
  Object {
    "createdAt": 2022-06-23T18:34:25.000Z,
    "id": 20,
    "name": "another test account",
    "updatedAt": 2022-06-23T19:38:01.000Z,
  },
]
`;
